compiler/rustc_metadata/src/rmeta/mod.rs

```diff
@@ -386,6 +386,7 @@ define_tables! {
     is_type_alias_impl_trait: Table<DefIndex, bool>,
     type_alias_is_lazy: Table<DefIndex, bool>,
     attr_flags: Table<DefIndex, AttrFlags>,
+    def_kind: Table<DefIndex, DefKind>,
```

Comment:
Why is it preferable to put it into defaulted set?

Could we just unwrap the option and keep it in the optional set, or do they behave differently for perf?

Reply:
I just didn't like having an `Option` in the table when the def kind is not actually optional, there should be no perf difference here.
But it may be marginally cleaner (and less changes) to keep the `Option`, yes.

Reply:
I think I would prefer the optional one, though I guess not particularly strongly, since it means we can go without the `Default` impl, for example.

Reply:
Reverted this change.